akka-actor/src/main/scala/akka/pattern/Patterns.scala

```diff
@@ -328,9 +336,13 @@ object PatternsCS {
    *   },
    *   timeout);
    * }}}
+   *
+   * @param actor          the actor to be asked
+   * @param messageFactory function taking an actor ref and returning the message to be sent
+   * @param timeout        the timeout for the response before failing the returned completion stage
    */
-  def ask(actor: ActorRef, messageFactory: japi.Function[ActorRef, Any], timeout: Timeout): CompletionStage[AnyRef] =
-    scalaAsk(actor, messageFactory.apply _)(timeout).toJava.asInstanceOf[CompletionStage[AnyRef]]
+  def explicitAsk(actor: ActorRef, messageFactory: japi.function.Function[ActorRef, Any], timeout: Timeout): CompletionStage[AnyRef] =
```

Comment:
Not sure about the name, how is an ask with a factory more "explicit" than an ask with just a message? What about `askWithActorRef`?

Reply:
I agree. Actually I simply took the name of the trait which provides this feature. Should we rename that, too?

Reply:
I don't find the trait name that important as that is internal and pretty simple/isolated.